addon/test-helper/deprecation.js

```diff
@@ -1,53 +1,50 @@
-/* globals QUnit */
-
-import MethodCallExpectation from "./method-call-expectation";
-
-var NONE = function(){};
+import MethodCallExpectation from './method-call-expectation';
+import MethodCallTracker from './method-call-tracker';
 
 var DeprecationAssert = function(env){
   this.env = env;
-
   this.reset();
 };
 
 DeprecationAssert.prototype = {
-
   reset: function(){
-    this.expecteds = null;
-    this.actuals = null;
+    this.tracker = new MethodCallTracker(this.env.Ember, 'deprecate');
```

Comment:
Given the way that the helpers behave when passed a callback vs. not, I thought this turned out to be a decent abstraction, although I see why it can seem weird. Really, it just fell out of extracting the tracking logic into a util, and resulted in the removal of the actual/expected swapping logic that was pretty difficult to follow IMO.

I think the alternative to creating new instances of the tracker would be to give them something like a `createNewContext` method so that the util could internally manage the temporary context shift. I have a feeling you were thinking something specific?